numba/cuda/tests/cudapy/test_cuda_array_interface.py

```diff
@@ -255,6 +256,154 @@ def test_consuming_strides(self):
         np.testing.assert_array_equal(got, hostarray)
         self.assertTrue(got.flags['C_CONTIGUOUS'])
 
+    def test_produce_no_stream(self):
+        c_arr = cuda.device_array(10)
+        self.assertIsNone(c_arr.__cuda_array_interface__['stream'])
+
+    def test_produce_stream(self):
+        s = cuda.stream()
+        c_arr = cuda.device_array(10, stream=s)
+        cai_stream = c_arr.__cuda_array_interface__['stream']
+        self.assertEqual(s.handle.value, cai_stream)
+
+    def test_consume_no_stream(self):
+        # Create a foreign array with no stream
+        f_arr = ForeignArray(cuda.device_array(10))
+
+        # Ensure that the imported array has no default stream
+        c_arr = cuda.as_cuda_array(f_arr)
+        self.assertEqual(c_arr.stream, 0)
+
+    def test_consume_stream(self):
+        # Create a foreign array with a stream
+        s = cuda.stream()
+        f_arr = ForeignArray(cuda.device_array(10, stream=s))
+
+        # Ensure that an imported array has the stream as its default stream
+        c_arr = cuda.as_cuda_array(f_arr)
+        self.assertTrue(c_arr.stream.external)
+        self.assertEqual(c_arr.stream.handle.value, s.handle.value)
+
+    def test_consume_no_sync(self):
+        # Create a foreign array with no stream
+        f_arr = ForeignArray(cuda.device_array(10))
+
+        with patch.object(cuda.cudadrv.driver.Stream, 'synchronize',
+                          return_value=None) as mock_sync:
+            cuda.as_cuda_array(f_arr)
+
+        # Ensure the synchronize method of a stream was not called
+        mock_sync.assert_not_called()
+
+    def test_consume_sync(self):
+        # Create a foreign array with a stream
+        s = cuda.stream()
+        f_arr = ForeignArray(cuda.device_array(10, stream=s))
+
+        with patch.object(cuda.cudadrv.driver.Stream, 'synchronize',
+                          return_value=None) as mock_sync:
+            cuda.as_cuda_array(f_arr)
+
+        # Ensure the synchronize method of a stream was called
+        mock_sync.assert_called_once_with()
+
+    def test_consume_sync_disabled(self):
+        # Create a foreign array with a stream
+        s = cuda.stream()
+        f_arr = ForeignArray(cuda.device_array(10, stream=s))
+
+        # Set sync to false before testing. The test suite should generally be
+        # run with sync enabled, but stash the old value just in case it is
+        # not.
+        old_sync = config.CUDA_ARRAY_INTERFACE_SYNC
+        config.CUDA_ARRAY_INTERFACE_SYNC = False
```

Comment:
This is mutating state. Suggest using https://github.com/numba/numba/blob/69f99050e0fd9f012259b93734488698f0e73f91/numba/tests/support.py#L507-L519, such that if this test fails prior to resetting the variable, the config is reset and doesn't cause issues for other tests that might not have run yet.

Reply:
Ah, I had a feeling there was something like that but didn't end up looking for it - thanks for the pointer!